src/middlewares/logger.middleware.ts

```diff
@@ -0,0 +1,25 @@
+import { Injectable, Logger, NestMiddleware } from '@nestjs/common';
+import { Request, Response, NextFunction } from 'express';
+import { JwtService } from '@nestjs/jwt';
+
+@Injectable()
+export class LoggerMiddleware implements NestMiddleware {
+  private readonly logger = new Logger();
+  private readonly jwt = new JwtService();
+
+  use(req: Request, res: Response, next: NextFunction) {
+    const token = req.headers?.authorization?.split(' ')[1] || '';
+    const userDetail = this.jwt.decode(token) || {};
```

Comment:
PullProofAI

Ensure proper error handling for missing or invalid authorization headers.
```diff
+     const token = req.headers?.authorization?.split(' ')[1] || '';
+     const userDetail = this.jwt.decode(token) || {};
```



<!-- This is an auto-generated comment by PullProofAI -->